src/ophyd_async/panda/writers/hdf_writer.py

```diff
@@ -0,0 +1,234 @@
+                #   Original exception was:
+                PandaHDFWriter._ctxt = None
+
+        return PandaHDFWriter._ctxt
+
+    def __init__(
+        self,
+        prefix: str,
+        directory_provider: DirectoryProvider,
+        name_provider: NameProvider,
+        panda_device: PandA,
+    ) -> None:
+        self.panda_device = panda_device
+        self._prefix = prefix
+        self._directory_provider = directory_provider
+        self._name_provider = name_provider
+        self._datasets: List[_HDFDataset] = []
+        self._file: Optional[_HDFFile] = None
+
+        # TODO add panda.data to the Panda device instead, as a typed block
+        self.hdf = HdfSignals(
+            panda_device.data.hdfdirectory,
+            panda_device.data.hdffilename,
+            panda_device.data.numcapture,
+            panda_device.data.numcaptured,
+            panda_device.data.capture,
+        )
+
+        # Get capture PVs by looking at panda. Gives mapping of dotted attribute path
+        # to Signal object
+        self.capture_signals = get_capture_signals(self.panda_device)
+
+    # Triggered on PCAP arm
+    async def open(self, multiplier: int = 1) -> Dict[str, Descriptor]:
+        """Retrieve and get descriptor of all PandA signals marked for capture"""
+
+        # Ensure flushes are immediate
+        await self.panda_device.data.flushperiod.set(0)
+
+        self.to_capture = await get_signals_marked_for_capture(self.capture_signals)
+        self._file = None
+        info = self._directory_provider()
+        await asyncio.gather(
+            self.hdf.file_path.set(info.directory_path),
+            self.hdf.file_name.set(f"{info.filename_prefix}.h5"),
+        )
+
+        # TODO confirm all missing functionality from AD writer isn't needed here
+
+        await self.hdf.num_capture.set(0)
+        # Wait for it to start, stashing the status that tells us when it finishes
+        await self.hdf.capture.set(True)
+        name = self._name_provider()
+        if multiplier > 1:
+            raise ValueError(
+                "All PandA datasets should be scalar, multiplier should be 1"
+            )
+        self._datasets = []
+        for attribute_path, value in self.to_capture.items():
+            # TODO check that a 'abc_capture' signal always records an 'abc_val' signal
+            signal_name = attribute_path.split(".")[-1]
+            block_name = attribute_path.split(".")[-2]
+
+            # Get block names from numbered blocks, eg INENC[1]
+            if block_name.isnumeric():
+                actual_block = attribute_path.split(".")[-3]
+                block_name = f"{actual_block}.{block_name}"
+
+            if value["capture_type"] == Capture.MinMaxMean:
+                shape = [3]
+            elif value["capture_type"] == Capture.MinMax:
+                shape = [2]
+            else:
+                shape = [1]
+                # TODO check if this is correct format
```

Comment:
Actually `Min Max Mean` will generate 3 datasets, so you need something like:
```python
for suffix in value["capture_type"].value.split(" "):
    self._datasets.append(
        _HDFDataset(...., path=f"{actual_block}{block_name}.{signal_name}.{suffix}", ...)
```

Reply:
Actually it strikes me that all this information is available in the capture table:
https://github.com/PandABlocks/PandABlocks-ioc/blob/acfefb3da61db47a32d6fff71f26236c35909812/src/pandablocks_ioc/_pvi.py#L183-L227

@evalott100 @olliesilvester I think we need a discussion about the PVI structures of PandA and how it will fit together